isis/src/base/apps/crop/crop.cpp

```diff
@@ -1,35 +1,37 @@
-#include "Isis.h"
-
 #include <cmath>
 
-#include "Cube.h"
-#include "ProcessByLine.h"
-#include "SpecialPixel.h"
-#include "LineManager.h"
-#include "FileName.h"
-#include "IException.h"
-#include "Projection.h"
-#include "AlphaCube.h"
-#include "Table.h"
-#include "SubArea.h"
+#include "crop.h"
 
 using namespace std;
 using namespace Isis;
 
-// Globals and prototypes
-int ss, sl, sb;
-int ns, nl, nb;
-int sinc, linc;
-Cube *cube = NULL;
-LineManager *in = NULL;
+PvlGroup crop(UserInterface &ui) {
+  // Globals and prototypes
+  int ss, sl, sb;
+  int ns, nl, nb;
+  int sinc, linc;
+
+  Cube *cube = NULL;
+  LineManager *in = NULL;
+
+  // Line processing routine
+  auto cropProccess = [&](Buffer &out)->void {
+    // Read the input line
+    int iline = sl + (out.Line() - 1) * linc;
+    in->SetLine(iline, sb);
+    cube->read(*in);
+
+    // Loop and move appropriate samples
+    for(int i = 0; i < out.size(); i++) {
+      out[i] = (*in)[(ss - 1) + i * sinc];
+    }
```

Comment:
ProccessByLine/Brick can use lambdas with captures, this means the functions can be defined within the app func while capturing the required additional values as long as they are defined in the scope of the function. This avoids multiple packing/unpacking from using a UI object (which doesn't support arbitrary types) or void* array.